lib/completion-templates.ts

```diff
@@ -5,45 +5,47 @@ export const completionShTemplate = `###-begin-{{app_name}}-completions-###
 # Installation: {{app_path}} {{completion_command}} >> ~/.bashrc
 #    or {{app_path}} {{completion_command}} >> ~/.bash_profile on OSX.
 #
-_{{app_name}}_yargs_completions()
+_{{safe_app_name}}_yargs_completions()
 {
-    local cur_word args type_list
+    if [[ {{app_name_word_check}} ]]; then
+      local cur_word args type_list
 
-    cur_word="\${COMP_WORDS[COMP_CWORD]}"
-    args=("\${COMP_WORDS[@]}")
+      cur_word="\${COMP_WORDS[COMP_CWORD]}"
+      args=("\${COMP_WORDS[@]}")
 
-    # ask yargs to generate completions.
-    type_list=$({{app_path}} --get-yargs-completions "\${args[@]}")
+      # ask yargs to generate completions.
+      type_list=$({{app_path}} --get-yargs-completions "\${args[@]}")
 
-    COMPREPLY=( $(compgen -W "\${type_list}" -- \${cur_word}) )
+      COMPREPLY=( $(compgen -W "\${type_list}" -- \${cur_word}) )
 
-    # if no match was found, fall back to filename completion
-    if [ \${#COMPREPLY[@]} -eq 0 ]; then
-      COMPREPLY=()
+      # if no match was found, fall back to filename completion
+      if [ \${#COMPREPLY[@]} -eq 0 ]; then
+        COMPREPLY=()
+      fi
     fi
 
     return 0
 }
-complete -o bashdefault -o default -F _{{app_name}}_yargs_completions {{app_name}}
+complete -o bashdefault -o default -F _{{safe_app_name}}_yargs_completions {{app_name_first_word}}
```

Comment:
`complete` doesn't have a way of handling multiple words at once, so we have to just give it the first word to kick of the process. The function itself checks if it will need to set the "`COMPREPLY`" or not.